### PR TITLE
Speedup

### PR DESCRIPTION
**Pull request by j-po**
I should probably clean up some of the profiling I vomited onto this, but this should reduce by just over 50% our calls to `candidate-badness`.

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -6,7 +6,8 @@
   :jvm-opts ["-Djava.awt.headless=true"]
   :dependencies [[org.clojure/clojure "1.8.0"]
                  [org.clojure/tools.cli "0.3.5"]
-                 [net.mikera/imagez "0.10.0"]]
+                 [net.mikera/imagez "0.10.0"]
+		 [com.taoensso/tufte "1.0.2"]]
```
Comment (GregorStocks) [inline project.clj]: you seem to be using tab characters, which i disapprove of, and you also seem to be using 8.5-space tabs, which i've got no problem with

**Comment by GregorStocks** [comment]
LGTM apart from the tab characters. That said, most of this diff is the profiling stuff, which we don't need most of the time - might make sense to pull out the new test and the code change into a PR that you land on master, and leave the profiling stuff separately on a branch which you can temporarily merge in when you actually need to profile stuff again.

File: project.clj (hunk 1 above)
Comment (j-po) [inline project.clj]: I, uh, did not notice that. #willfix

**Comment by j-po** [comment]
The new test requires all of the profiling vomit, sadly, and I'm guessing
it would take some doing to make said vomit not a nightmarish merge every
time. Maybe putting everything on its own line would be enough, though.

**Pull request merged**